packages/vite/src/node/ssr/__tests__/ssrTransform.spec.ts

```diff
@@ -932,3 +932,17 @@ const Baz = class extends Foo {}
     "
   `)
 })
+
+test('import assertion attribute', async () => {
+  expect(
+    await ssrTransformSimpleCode(`
+  import * as foo from './foo.json' assert { type: 'json' };
+  import('./bar.json', { assert: { type: 'json' } });
+  `),
+  ).toMatchInlineSnapshot(`
+  "const __vite_ssr_import_0__ = await __vite_ssr_import__(\\"./foo.json\\");
+  
+    
+    __vite_ssr_dynamic_import__('./bar.json', { assert: { type: 'json' } });
```

Comment:
Maybe `__vite_ssr_dynamic_import__ `  and `__vite_ssr_import__` should throw an error if the runtime load a non-json module for the case. What do you think cc @https://github.com/sheremet-va

Reply:
vite now will parse the json file's schema in the runtime, I think we could add some assertion validations in the parsing phase:

1. only `json` assert type is allowed like https://nodejs.org/api/esm.html#import-assertions said
2. module's filepath suffix must be `.json` when assert type is `json`

Reply:
If you really want to support Node.js heuristics, there are more checks with assertion:

- https://github.com/nodejs/node/blob/main/lib/internal/modules/esm/assert.js

Reply:
> If you really want to support Node.js logic, there are more checks with assertion:
> 
> * https://github.com/nodejs/node/blob/main/lib/internal/modules/esm/assert.js

IIUC, only `json` assert type is supported for now according to 

https://github.com/nodejs/node/blob/56c3263049dd75527155418e8f7f3a2bc2709f17/lib/internal/modules/esm/assert.js#L19-L45

IMO, a simple approach is  that checking whether the import statement matches the pattern like `import xxx from '*.json'  assert { type: 'json' }` or ` import('*.json', { assert: { type: 'json' } })`.

Another approach is that we could directly pass assert type to `dynamicImport` to reuse node builtin `import()`. It's more robust with more complicated implementation.

https://github.com/vitejs/vite/blob/91a18c2f7da796ff8217417a4bf189ddda719895/packages/vite/src/node/ssr/ssrModuleLoader.ts#L298-L299

Reply:
I think it already errors if it's a non-json module because Vite will try to convert to json to a js file and fails. But nonetheless if it's an issue, I don't think it's worthing working more on it as import assertions is already deprecated.